fastlane/Fastfile

```diff
@@ -770,6 +776,51 @@ end
       UI.user_error!("This lane should be run only from CI")
     end
   end
+
+  desc 'Registers a Device in the developer console'
+  lane :register_new_device do |options|
+    device_name = UI.input('Device Name: ') if options[:device_name].nil?
+    device_id = UI.input('Device ID: ') if options[:device_id].nil?
+    all_bundle_ids = simplenote_app_identifiers
+
+    UI.message "Registering #{device_name} with ID #{device_id} and registering it with any provisioning profiles associated with these bundle identifiers:"
+    all_bundle_ids.each do |identifier|
+      UI.message "\t#{identifier}"
+    end
+
+    team_id = get_required_env('EXT_EXPORT_TEAM_ID')
+
+    # Register the user's device
+    register_device(
+      name: device_name,
+      udid: device_id,
+      team_id: team_id,
+      api_key_path: APP_STORE_CONNECT_KEY_PATH
+    )
+
+    # We're about to use `add_development_certificates_to_provisioning_profiles` and `add_all_devices_to_provisioning_profiles`.
+    # These actions use Developer Portal APIs that don't yet support authentication via API key (-.-').
+    # Let's preemptively ask for and set the email here to avoid being asked twice for it if not set.
+
+    require 'credentials_manager'
+
+    # If Fastlane cannot instantiate a user, it will ask the caller for the email.
+    # Once we have it, we can set it as `FASTLANE_USER` in the environment (which has lifecycle limited to this call) so that the next commands will already have access to it.
+    # Note that if the user is already available to `AccountManager`, setting it in the environment is redundant, but Fastlane doesn't provide a way to check it so we have to do it anyway.
+    ENV['FASTLANE_USER'] = CredentialsManager::AccountManager.new.user
+
+    # Add all development certificates to the provisioning profiles (just in case – this is an easy step to miss)
+    add_development_certificates_to_provisioning_profiles(
+      team_id: team_id,
+      app_identifier: all_bundle_ids
+    )
+
+    # Add all devices to the provisioning profiles
+    add_all_devices_to_provisioning_profiles(
+      team_id: team_id,
+      app_identifier: all_bundle_ids
+    )
```

Comment:
Worth noting that, when I run the automation, these two printed warnings about not finding profiles matching the given identifiers:

<img width="1432" alt="image" src="https://user-images.githubusercontent.com/1218433/200632148-3725caea-1c91-4cf8-8aca-490275ddc416.png">

This is actually the correct behavior, because Simplenote uses a different bundle identifier with an additional `Development` reverse-DNS component for the development builds.